test/jdk/java/nio/channels/Channels/TransferTo.java

```diff
@@ -141,24 +141,29 @@ public void testMoreThanTwoGB() throws IOException {
         // preparing two temporary files which will be compared at the end of the test
         Path sourceFile = Files.createTempFile(null, null);
         Path targetFile = Files.createTempFile(null, null);
+        try {
+            // writing 3 GB of random bytes into source file
+            for (int i = 0; i < NUM_WRITES; i++)
+                Files.write(sourceFile, createRandomBytes(BYTES_PER_WRITE, 0), StandardOpenOption.APPEND);
+
+            // performing actual transfer, effectively by multiple invocations of Filechannel.transferTo(FileChannel)
+            long count;
+            try (InputStream inputStream = Channels.newInputStream(FileChannel.open(sourceFile));
+                 OutputStream outputStream = Channels
+                         .newOutputStream(FileChannel.open(targetFile, StandardOpenOption.WRITE))) {
+                count = inputStream.transferTo(outputStream);
+            }
 
-        // writing 3 GB of random bytes into source file
-        for (int i = 0; i < NUM_WRITES; i++)
-            Files.write(sourceFile, createRandomBytes(BYTES_PER_WRITE, 0), StandardOpenOption.APPEND);
+            // comparing reported transferred bytes, must be 3 GB
+            assertEquals(count, BYTES_WRITTEN);
 
-        // performing actual transfer, effectively by multiple invocations of Filechannel.transferTo(FileChannel)
-        long count;
-        try (InputStream inputStream = Channels.newInputStream(FileChannel.open(sourceFile));
-                OutputStream outputStream = Channels
-                        .newOutputStream(FileChannel.open(targetFile, StandardOpenOption.WRITE))) {
-            count = inputStream.transferTo(outputStream);
+            // comparing content of both files, failing in case of any difference
+            assertEquals(Files.mismatch(sourceFile, targetFile), -1);
+        } finally {
+            System.out.printf("Cleaning up: %s, %s%n", sourceFile, targetFile);
+            Files.deleteIfExists(sourceFile);
+            Files.deleteIfExists(targetFile);
```

Comment:
What about an exception thrown from `Files.deleteIfExists(sourceFile)` in that `finally` block?

Reply:
Right - maybe that should be done in a private method where the exception would be catch'ed and logged (`printStackTrace`) - otherwise an exception in the finally clause risks masking any exception that was thrown before the finally clause was reached.